applauncher/004-three-by-three.qml

```diff
@@ -0,0 +1,358 @@
+        }
+    }
+
+    MouseArea {
+        id: absoluteMouse
+
+        anchors.fill: parent
+        propagateComposedEvents: true
+
+        onPressed: {
+            clickY = Math.round(mouse.y)
+            mouse.accepted = false
+        }
+        onClicked: mouse.accepted = false
+        onReleased: mouse.accepted = false
+        onDoubleClicked: mouse.accepted = false
+        onPositionChanged: mouse.accepted = false
+        onPressAndHold: mouse.accepted = false
+    }
+
+    Text {
+        id: hoverTitle
+
+        property string tempTitle: appTitle
+        property real hoverTextOffset: root.clickY > Dims.h(48) ? -Dims.h(62) : Dims.h(62)
+        property bool rootPressToggle: root.pressToggle
+
+        z: 1
+        width: parent.width
+        color: "#ffffff"
+        opacity: 0
+        visible: !root.clickToggle
+        horizontalAlignment: Text.AlignHCenter
+        anchors.centerIn: parent
+        style: Text.Outline;
+        styleColor: alb.centerColor(launcherModel.get(currentPressedIndex).filePath)
+        font {
+            pixelSize: ((appsView.width > appsView.height ? appsView.height : appsView.width) / Dims.l(100)) * Dims.l(8)
+            styleName: "Condensed Medium"
+            letterSpacing: -parent.width * .002
+        }
+        text: appTitle.toUpperCase() + localeManager.changesObserver
+
+        Behavior on rootPressToggle {
+            SequentialAnimation {
+                id: fadeText
+
+                NumberAnimation { target: hoverTitle; property: "anchors.verticalCenterOffset"; to: hoverTitle.hoverTextOffset; duration: 0}
+                NumberAnimation { target: hoverTitle; property: "opacity"; to: 1; duration: 0}
+                NumberAnimation { target: titleShutter; property: "opacity"; to: .85; duration: 0}
+
+                PropertyAction{}
+
+                NumberAnimation { target: hoverTitle; property: "anchors.verticalCenterOffset"; to: -hoverTitle.hoverTextOffset + (hoverTitle.hoverTextOffset * 1.62); duration: 100; easing.type: Easing.InSine}
+
+                PauseAnimation { duration: 800 }
+
+                ParallelAnimation {
+
+                    NumberAnimation { target: hoverTitle; property: "opacity"; to: 0; duration: 200; easing.type: Easing.InSine}
+                    NumberAnimation { target: titleShutter; property: "opacity"; to: 0; duration: 200; easing.type: Easing.InSine}
+                    NumberAnimation { target: hoverTitle; property: "anchors.verticalCenterOffset"; to: hoverTitle.hoverTextOffset; duration: 200; easing.type: Easing.InSine}
+                }
+            }
+        }
+
+        layer.enabled: true
+        layer.effect: DropShadow {
+            transparentBorder: true
+            horizontalOffset: 0
+            verticalOffset: 0
+            radius: 2.0
+            samples: 5
+            color: "#88000000"
+        }
+    }
+
+    Rectangle {
+        id: titleShutter
+
+        z: 0
```

Comment:
If you can avoid manually specifying z orders by using the natural ordering of the QML file, I think it’s best.
Z orders always feel a bit hacky and hard to read, it should be avoided in situations where it can be

Reply:
I think i added z when i ran into an instance where items overlapped the titleShutter. But will try without now. The overlap might ave been not solved by the z even.